client/src/thin_client.rs

```diff
@@ -118,64 +123,52 @@ impl ClientOptimizer {
 }
 
 /// An object for querying and sending transactions to the network.
-pub struct ThinClient<C: 'static + TpuConnection> {
+pub struct ThinClient {
     rpc_clients: Vec<RpcClient>,
-    tpu_connections: Vec<C>,
+    tpu_addrs: Vec<SocketAddr>,
```

Comment:
I don't _think_ there are (m?)any external consumers of  `ThinClient`, but it _is_ a published, public interface and we _do_ have a [backward compatibility policy](https://docs.solana.com/developing/backwards-compatibility) to respect

Reply:
Do we want add the UdpSocket param back in but mark it as deprecated and note that it will be ignored? Or is it fine as-is?

Reply:
I _think_ we'll be fine as is for this one.  Worst case we learn who the third-party consumers are and tell them they're doing it wrong.  This note was more to raise awareness for future changes